front/src/components/Main/PostList.js

```diff
@@ -1,17 +1,70 @@
-import Pagination from '@mui/material/Pagination';
 import * as React from 'react';
-import Stack from '@mui/material/Stack';
+import {useSearchParams} from 'react-router-dom';
 import Post from './Post';
+import LoadMainPage from '../../features/LoadMainPage';
+import Previous from '../entrie/Previous';
+import Next from '../entrie/Next';
+
+export default function PostList(a) {
+  const [data, setData] = React.useState([]);
+  const [wholePage, setWholePage] = React.useState();
+  const [page, setPage] = React.useState(1);
+  const [searchParams] = useSearchParams();
+  const [category, setCategory] = React.useState(0);
+
+  React.useEffect(() => {
```

Comment:
```suggestion
import { useEffect } from 'react';
  useEffect(() => {
```
앞에 React를 매 번 붙이는게 아니라, 구조분해 할당으로 불러온 값을 통해서 useEffect만 사용할 수 있습니다.